### PR TITLE
Added ruby snippets

### LINKED ISSUE #249: Create code snippets for homepage (ruby)
**What**

We need to create Ruby code snippets for the ably homepage as outlined in [https://ably.atlassian.net/browse/SDK-878](https://ably.atlassian.net/browse/SDK-878|smart-link) .

The Ruby snippets needed are:

* Chat (publish)
* GPS (publish)
* IoT (publish and subscribe)
* Live Updates (publish)
* Multi-user Spaces (publish)

Please refer to the existing JavaScript snippets:

[https://github.com/ably/homepage-snippets/tree/main/javascript](https://github.com/ably/homepage-snippets/tree/main/javascript|smart-link) 

**Acceptance criteria**

A PR should be raised in the homepage-snippets repository: [https://github.com/ably/homepage-snippets/pulls](https://github.com/ably/homepage-snippets/pulls)

┆Issue is synchronized with this [Jira Uncategorised](https://ably.atlassian.net/browse/SDK-881) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by ghost**
Fixes https://github.com/ably/ably-ruby/issues/249

### CODE REVIEW

File: ruby/chat_pub.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+require 'ably'
```
Comment (mattheworiordan) [inline ruby/chat_pub.rb]: The JS version doesn't include necessary constructor code, so this + ably = Ably::Realtime.new(...) is not necessary. I believe this is noise, we only want to show the code that is relevant

Comment (mattheworiordan) [reply ruby/chat_pub.rb]: I see that we don't yet have finalisation of the JS version though, see https://github.com/ably/homepage-snippets/pull/1.  I appreciate this makes this job harder, but the goal should be to have as close an equivalent, that is terse.  Please can you be more terse and only show what is relevant to the visitor to Ably.  Specifically, see https://github.com/ably/homepage-snippets/pull/1#pullrequestreview-684069614, and comment:

> 1. Show how simple Ably is
> 2. Show real code that could actually works (at least the extracted part that we're showing)
> 3. Hide irrelevant boilerplate code.

File: ruby/gps_pub.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+require 'ably'
+
+lat, long = 10.1233, 12.22
```
Comment (mattheworiordan) [inline ruby/gps_pub.rb]: No need to show the values.  In fact, this makes the code sample more questionable given that's hard coded.

File: ruby/gps_pub.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+
+ably = Ably::Realtime.new(key: ENV['ABLY_API_KEY'])
+channel = ably.channels.get 'driver:elon'
```
Comment (mattheworiordan) [inline ruby/gps_pub.rb]: This style doesn't align with our Ruby style in our SDK.  This should use parenthesis, `ably.channels.get('driver:elon')`. See https://github.com/ably/ably-ruby

**Review by mattheworiordan — CHANGES_REQUESTED** [review]
Please see my comments at https://github.com/ably/homepage-snippets/pull/1#pullrequestreview-684069614

File: ruby/gps_pub.rb (hunk 2 above)
Comment (ghost) [reply ruby/gps_pub.rb]: Fixed

File: ruby/gps_pub.rb (hunk 3 above)
Comment (ghost) [reply ruby/gps_pub.rb]: Fixed

File: ruby/chat_pub.rb (hunk 1 above)
Comment (ghost) [reply ruby/chat_pub.rb]: Fixed

**Comment by lmars** [comment]
@lukaszsliwa apologies there's been some confusion around which Ruby snippets are required.

I've updated [the Jira ticket](https://ably.atlassian.net/browse/SDK-881) with an explicit list, which I'll also copy here:

The Ruby snippets needed are:

- Chat (publish)
- GPS (publish)
- IoT (publish and subscribe)
- Live Updates (publish)
- Multi-user Spaces (publish)

Please drop any snippets you've added which aren't in that list, and please bring the remaining ones inline with [the JavaScript snippets](https://github.com/ably/homepage-snippets/tree/main/javascript).

**Comment by ghost** [comment]
@lmars Thanks for the message. Updated https://github.com/ably/homepage-snippets/pull/2/commits/bfb5009cc0070c2b4f38f52b5e50b6fdb0f87e11

File: ruby/chat_pub.rb [hunk 4]
```diff
@@ -0,0 +1,3 @@
+channel = ably.channels.get('room:tesla-fans')
+channel.presence.enter_client('client', 'metadata')
```
Comment (lmars) [inline ruby/chat_pub.rb]: This should simply be `channel.presence.enter` like the JavaScript snippet (which I've just fixed in https://github.com/ably/homepage-snippets/pull/11)

File: ruby/chat_sub.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+channel = ably.channels.get('room:tesla-fans')
```
Comment (lmars) [inline ruby/chat_sub.rb]: Please delete this snippet as it is not required for Ruby

File: ruby/gps_sub.rb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+channel = ably.channels.get('driver:elon')
```
Comment (lmars) [inline ruby/gps_sub.rb]: Please delete this snippet as it is not required for Ruby

File: ruby/iot_sub.rb [hunk 7]
```diff
@@ -0,0 +1,6 @@
+mqtt = ably.channels.get('device:rocket:1')
+mqtt.subscribe('message') do |payload|
```
Comment (lmars) [inline ruby/iot_sub.rb]: This snippet should demonstrate using an MQTT client to connect to Ably like the [equivalent JavaScript snippet](https://github.com/ably/homepage-snippets/blob/main/javascript/iot-sub.js), for example using the [mqtt gem](https://github.com/njh/ruby-mqtt):

```ruby
mqttClient.subscribe('device:rocket:1')
mqttClient.get do |topic, message|
  if topic == 'action'
    rocket.process_instruction(JSON.parse(message))
  end
end
```

File: ruby/live_updates_sub.rb [hunk 8]
```diff
@@ -0,0 +1,4 @@
+channel = ably.channels.get('dogecoin:usd')
```
Comment (lmars) [inline ruby/live_updates_sub.rb]: Please delete this snippet as it is not required for Ruby

File: ruby/multi_user_sub.rb [hunk 9]
```diff
@@ -0,0 +1,4 @@
+channel = ably.channels.get('doc:mars-launch-plan')
```
Comment (lmars) [inline ruby/multi_user_sub.rb]: Please delete this snippet as it is not required for Ruby

**Review by lmars — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
@lmars, thanks for the comments.

Fixed in https://github.com/ably/homepage-snippets/pull/2/commits/f904c85b87e5ce80d5fb4176eafb7c2a37e71f0c

**Review by lmars — APPROVED** [review]

**Comment by lmars** [comment]
@lukaszsliwa Thanks!

**Pull request merged**